Dev server cannot start in monorepo project

Hello!

I tried to setup Flareact in monorepo project, but `flareact dev` is always fail. From the error being printed, `webpack-dev-server `cannot resolve `node_modules/flareact/configs/webpack.client.config.js`.

My Flareact site is located on `/website`, while the real `node_modules` is located on `/`. But `webpack-dev-server` is lookin for the config on `/website/node_modules`.

Here are my environments:
- Node version: v14.15.3
- Package manager: Yarn v1.22.10
- OS: Ubuntu 20.04 on WSL2

Error message:
```
🚀 Starting Flareact dev server on http://localhost:8080 ...

[client] internal/modules/cjs/loader.js:883
[client]   throw err;
[client]   ^
[client] 
[client] Error: Cannot find module '/home/izznatsir/projects/github.com/izznatsir/radix-tw/website/node_modules/flareact/configs/webpack.client.config.js'
[client] Require stack:
[client] - /home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js
[client] - /home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-dev-server/bin/webpack-dev-server.js
[client]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
[client]     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
[client]     at Module.require (internal/modules/cjs/loader.js:952:19)
[client]     at require (internal/modules/cjs/helpers.js:88:18)
[client]     at WEBPACK_OPTIONS (/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js:114:13)
[client]     at requireConfig (/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js:116:6)
[client]     at /home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js:123:17
[client]     at Array.forEach (<anonymous>)
[client]     at module.exports (/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js:121:15)
[client]     at Object.<anonymous> (/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-dev-server/bin/webpack-dev-server.js:84:40) {
[client]   code: 'MODULE_NOT_FOUND',
[client]   requireStack: [
[client]     '/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-cli/bin/utils/convert-argv.js',
[client]     '/home/izznatsir/projects/github.com/izznatsir/radix-tw/node_modules/webpack-dev-server/bin/webpack-dev-server.js'
[client]   ]
[client] }
[client] webpack-dev-server --config node_modules/flareact/configs/webpack.client.config.js --mode development exited with code 1
```

--- a/src/bin/flareact.js
+++ b/src/bin/flareact.js
@@ -1,6 +1,5 @@
 #!/usr/bin/env node
 const fs = require("fs");
-const path = require("path");
 const concurrently = require("concurrently");
 const dotenv = require("dotenv");
 dotenv.config();
@@ -17,12 +16,11 @@ dotenv.config();
 
 const yargs = require("yargs");
 
-const rootPath = "";
+let rootPath = "";
 const webpackConfigPath =
   "node_modules/flareact/configs/webpack.client.config.js";
 
-let isWebpackConfigFound = () =>
-  fs.existsSync(path.resolve(__dirname, rootPath, webpackConfigPath));
+let isWebpackConfigFound = () => fs.existsSync(rootPath + webpackConfigPath);
 
 while (!isWebpackConfigFound()) {
   rootPath += "../";

--- a/src/bin/flareact.js
+++ b/src/bin/flareact.js
@@ -1,4 +1,6 @@
 #!/usr/bin/env node
+const fs = require("fs");
+const path = require("path");
 const concurrently = require("concurrently");
 const dotenv = require("dotenv");
 dotenv.config();
@@ -14,6 +16,17 @@ dotenv.config();
 });
 
 const yargs = require("yargs");
+
+const rootPath = "";
+const webpackConfigPath =
+  "node_modules/flareact/configs/webpack.client.config.js";
+
+let isWebpackConfigFound = () =>
+  fs.existsSync(path.resolve(__dirname, rootPath, webpackConfigPath));
+
+while (!isWebpackConfigFound()) {
+  rootPath += "../";
+}
 
 const argv = yargs
   .command("dev", "Starts a Flareact development server")
@@ -45,8 +58,9 @@ if (argv._.includes("dev")) {
         env: { WORKER_DEV: true, IS_WORKER: true },
       },
       {
-        command:
-          "webpack-dev-server --config node_modules/flareact/configs/webpack.client.config.js --mode development",
+        command: `webpack-dev-server --config ${
+          rootPath + webpackConfigPath
+        } --mode development`,
         name: "client",
         env: { NODE_ENV: "development" },
       },
@@ -78,7 +92,9 @@ if (argv._.includes("publish")) {
   concurrently(
     [
       {
-        command: `webpack --config node_modules/flareact/configs/webpack.client.config.js --out ./out --mode production && ${wranglerPublish}`,
+        command: `webpack --config ${
+          rootPath + webpackConfigPath
+        } --out ./out --mode production && ${wranglerPublish}`,
         name: "publish",
         env: { NODE_ENV: "production", IS_WORKER: true },
       },
@@ -102,8 +118,9 @@ if (argv._.includes("build")) {
   concurrently(
     [
       {
-        command:
-          "webpack --config node_modules/flareact/configs/webpack.client.config.js --out ./out --mode production",
+        command: `webpack --config ${
+          rootPath + webpackConfigPath
+        } --out ./out --mode production`,
         name: "publish",
         env: { NODE_ENV: "production" },
       },

--- a/src/bin/flareact.js
+++ b/src/bin/flareact.js
@@ -17,13 +17,29 @@ dotenv.config();
 const yargs = require("yargs");
 
 let rootPath = "";
-const webpackConfigPath =
+let webpackConfigPath =
   "node_modules/flareact/configs/webpack.client.config.js";
 
 let isWebpackConfigFound = () => fs.existsSync(rootPath + webpackConfigPath);
 
-while (!isWebpackConfigFound()) {
+for (let i = 0; i < 3; i++) {
+  if (isWebpackConfigFound()) {
+    break;
+  }
   rootPath += "../";
+}
+
+if (isWebpackConfigFound()) {
+  webpackConfigPath = rootPath + webpackConfigPath;
+} else {
+  const firstLine =
+    "⚠ Cannot find node_modules/flareact/configs/webpack.client.config.js.";
+  const secondLine =
+    "⚠ Make sure all your dependencies are installed. If your project is a monorepo,";
+  const thirdLine =
+    "⚠ make sure Flareact workspace is not nested more than 2 levels.";
+  console.error(firstLine + "\n" + secondLine + "\n" + thirdLine + "\n");
+  process.exit(1);
 }
 
 const argv = yargs
@@ -56,9 +72,7 @@ if (argv._.includes("dev")) {
         env: { WORKER_DEV: true, IS_WORKER: true },
       },
       {
-        command: `webpack-dev-server --config ${
-          rootPath + webpackConfigPath
-        } --mode development`,
+        command: `webpack-dev-server --config ${webpackConfigPath} --mode development`,
         name: "client",
         env: { NODE_ENV: "development" },
       },
@@ -90,9 +104,7 @@ if (argv._.includes("publish")) {
   concurrently(
     [
       {
-        command: `webpack --config ${
-          rootPath + webpackConfigPath
-        } --out ./out --mode production && ${wranglerPublish}`,
+        command: `webpack --config ${webpackConfigPath} --out ./out --mode production && ${wranglerPublish}`,
         name: "publish",
         env: { NODE_ENV: "production", IS_WORKER: true },
       },
@@ -116,9 +128,7 @@ if (argv._.includes("build")) {
   concurrently(
     [
       {
-        command: `webpack --config ${
-          rootPath + webpackConfigPath
-        } --out ./out --mode production`,
+        command: `webpack --config ${webpackConfigPath} --out ./out --mode production`,
         name: "publish",
         env: { NODE_ENV: "production" },
       },

--- a/src/bin/flareact.js
+++ b/src/bin/flareact.js
@@ -37,7 +37,7 @@ if (isWebpackConfigFound()) {
   const secondLine =
     "⚠ Make sure all your dependencies are installed. If your project is a monorepo,";
   const thirdLine =
-    "⚠ make sure Flareact workspace is not nested more than 2 levels.";
+    "⚠ make sure Flareact workspace is not nested more than 3 levels.";
   console.error(firstLine + "\n" + secondLine + "\n" + thirdLine + "\n");
   process.exit(1);
 }

